src/index.ts

```diff
@@ -6,18 +6,31 @@ import {
   StandaloneServerContextFunctionArgument,
   startStandaloneServer,
 } from "@apollo/server/standalone";
+import resolvers from "./resolvers";
+import { DataSourceContext } from "./types/DataSourceContext";
+import { GraphQLError } from "graphql";
 
 const port = process.env.PORT ?? "4001";
-import resolvers from "./resolvers";
 const subgraphName = require("../package.json").name;
-import { DataSourceContext } from "./types/DataSourceContext";
+const routerSecret = process.env.ROUTER_SECRET;
 
 const context: ContextFunction<
   [StandaloneServerContextFunctionArgument],
   DataSourceContext
-> = async ({ req }) => ({
-  auth: req.headers.authorization,
-});
+> = async ({ req }) => {
+  if (routerSecret && req.headers["router-authorization"] !== routerSecret) {
+    throw new GraphQLError("Missing router authentication", {
+      extensions: {
+        code: "UNAUTHENTICATED",
+        http: { status: 401 },
+      },
+    });
```

Comment:
wouldn't be best to use a plugin for this?

Reply:
I don't know, I don't really use Apollo Server 😅. I based this off of [these docs](https://www.apollographql.com/docs/apollo-server/security/authentication#api-wide-authorization), if there's a better way then we should use it! (and update the docs)